isolate_test.go

```diff
@@ -135,6 +135,68 @@ func TestIsolateGarbageCollection(t *testing.T) {
 	time.Sleep(time.Second)
 }
 
+func TestIsolateThrowException(t *testing.T) {
+	t.Parallel()
+	iso := v8.NewIsolate()
+	defer iso.Dispose()
+
+	strErr, err := v8.NewValue(iso, "some type error")
+
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+
+	throwError := func(val *v8.Value) {
+		v := iso.ThrowException(val)
+
+		if !v.IsNullOrUndefined() {
+			t.Error("expected result to be null or undefined")
+		}
+	}
+
+	// Function that throws a simple string error from within the function. It is meant
+	// to emulate when an error is returned within Go.
+	fn := v8.NewFunctionTemplate(iso, func(info *v8.FunctionCallbackInfo) *v8.Value {
+		throwError(strErr)
+
+		return nil
+	})
+
+	// Function that is passed a TypeError from JavaScript.
+	fn2 := v8.NewFunctionTemplate(iso, func(info *v8.FunctionCallbackInfo) *v8.Value {
+		args := info.Args()
+
+		if len(args) < 1 {
+			t.Error("expected a TypeError argument")
+		}
```

Comment:
Since the unit test is the only one using this function, we can probably assume that an argument is passed in. If we were testing the function template and info args behavior, it would make sense to check these but we cover that in `function_template_test.go`